### PR TITLE
feat(streams): log standardization debug stream

### PR DESCRIPTION
**Pull request by dm03514**
refs #SREPR-515

By submitting a PR to this repository, you agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md). Please see the [contributing guidelines](https://github.com/auth0/.github/blob/master/CONTRIBUTING.md) for how to create and submit a high-quality PR for this repo.

### Description

Provides a schema debug stream for checking logging against the standardized schema.  This will help mme instrument API2 by allowing me to toggle on this stream and then run the test suite.

### References

https://github.com/auth0/logs-schemas/releases/tag/v0.1.0

### Testing

> Describe how this can be tested by reviewers. Be specific about anything not tested and reasons why. If this library has unit and/or integration testing, tests should be added for new functionality and existing tests should complete without errors.
>
> Please include any manual steps for testing end-to-end or functionality not covered by unit/integration tests.
>
> Also include details of the environment this PR was developed in (language/platform/browser version).

- [ ] This change adds test coverage for new/changed/fixed functionality

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR or in auth0.com/docs
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [ ] The correct base branch is being used, if not `master`

# TODO
- [ ] Remove Unnecessary `delete` from instantiation
- [ ] Add configurable template

### CODE REVIEW

File: streams/standardization.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+const LOG_SCHEMAS = require('logs-schemas');
+
+
+class SchemaDebugStream extends Writable {
+    constructor(options) {
+        options = options || {};
+        super(options);
+        this.out = options.out || process.stderr;
+        delete options.out;
+
```
Comment (nw) [inline streams/standardization.js]: I like the defense but is delete really needed?

File: streams/standardization.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        const entry = chunk.toString();
+        let json;
+        try {
+            json = JSON.parse(entry);
+        } catch(error) {
+            return callback();
+        }
+
+        const result = this.v.validate(json, LOG_SCHEMAS.webservice);
+        if (!result.valid) {
```
Comment (nw) [inline streams/standardization.js]: should the LOG_SCHEMA type be configurable via options?

**Review by nw — COMMENTED** [review]
spoke w/ Danny about instrumentation integration. Current logging mechanism integration isn’t ideal, should be DI. However the footprint is small and should be a nonissue.

File: streams/standardization.js (hunk 1 above)
Comment (dm03514) [reply streams/standardization.js]: ohh nope, ha, the `super(options)` comes before 🙃

**Review by nw — APPROVED** [review]
LGTM

**Pull request merged**